cogniac/gateway.py

```diff
@@ -35,23 +50,88 @@ class CogniacGateway(object):
     media from another host on the same network.
     """
 
-    def __init__(self, timeout=60, url_prefix=None):
+    ##
+    #  get
+    ##
+    @classmethod
+    @retry(stop_max_attempt_number=8, wait_exponential_multiplier=500, retry_on_exception=server_error)
+    def get(cls,
+            connection=None,
+            gateway_id=None,
+            url_prefix=None):
+        """
+        get single network camera
+        connnection (CogniacConnection): Authenticated CogniacConnection object
+        gateway_id (String): the unique identifier of a Gateway object
+        returns CogniacGateway object
+        """
+        gateway_dict = None
+        if connection and gateway_id:
+            resp = connection._get("/gateways/{}".format(gateway_id))
+            gateway = resp.json()
+        return CogniacGateway(connection=connection, url_prefix=url_prefix, gateway_dict=gateway)
+
+    ##
+    #  get_all
+    ##
+    @classmethod
+    def get_all(cls, connection):
+        """
+        return all CogniacGateway objects belonging to the currently authenticated tenant
+
+        connnection (CogniacConnection):     Authenticated CogniacConnection object
+        """
+        resp = connection._get('/tenants/%s/gateways' % connection.tenant.tenant_id)
+        gateways = resp.json()['data']
+        return [CogniacGateway(connection=connection, gateway_dict=gateway) for gateway in gateways]
+
+    def __init__(self, connection=None, gateway_dict=None, timeout=60, url_prefix=None):
         """
         Initialize a CogniacGateway object.
         
         url_prefix (String):          URL prefix for a Cogniac EdgeFlow device.
         """
+        if gateway_dict:
+            super(CogniacGateway, self).__setattr__('_gateway_keys', gateway_dict.keys())
+
         if 'COG_GW_URL_PREFIX' in os.environ:
```

Comment:
CORE_FQDN

Reply:
Should I assume API version 1? Is there an environment variable for that or can we choose one, if not?

Reply:
`COG_GW_URL_PREFIX` actually refers to the EF's HTTP API, to enable API access to `/version`, `/process`, etc., so it's distinct from `CORE_FQDN` and `COG_URL_PREFIX`. There's probably a better name for it, though.

Reply:
to the extent this needs to be configured it should come from CloudCore not the environment.  Lets hardcode it to 1 for now and if it changes we will figure out how to communicate that via cloudcore config.

Reply:
I removed `COG_GW_URL_PREFIX` here. It refers to the address of the EdgeFlow's API, not CloudCore, which I believe isn't currently available via CloudCore.

Reply:
The IP address should be taken from the gateway object.